cognite_toolkit/cdf_tk/load.py

```diff
@@ -181,26 +210,19 @@ def dump(self, camel_case: bool = True) -> dict[str, Any]:
         }
 
 
-@dataclass
-class Difference:
-    added: list[CogniteResource]
-    removed: list[CogniteResource]
-    changed: list[CogniteResource]
-    unchanged: list[CogniteResource]
+class ExtractionPipelineConfigList(CogniteResourceList[ExtractionPipelineConfig]):
+    _RESOURCE = ExtractionPipelineConfig
 
-    def __iter__(self):
-        return iter([self.added, self.removed, self.changed, self.unchanged])
 
-    def __next__(self):
-        return next([self.added, self.removed, self.changed, self.unchanged])
+T_ID = TypeVar("T_ID", bound=Union[str, int, DataModelingId, InstanceId, VersionedDataModelingId, RawTable])
 
+T_CogniteResourceWrite = TypeVar("T_CogniteResourceWrite", bound=CogniteResource)
+T_CogniteResourceWriteList = TypeVar("T_CogniteResourceWriteList", bound=CogniteResourceList)
 
-T_ID = TypeVar("T_ID", bound=Union[str, int])
-T_Resource = TypeVar("T_Resource")
-T_ResourceList = TypeVar("T_ResourceList")
 
-
-class Loader(ABC, Generic[T_ID, T_Resource, T_ResourceList]):
+class Loader(
+    ABC, Generic[T_ID, T_CogniteResourceWrite, T_CogniteResource, T_CogniteResourceWriteList, T_CogniteResourceList]
```

Comment:
This results in all loaders seemingly duplicating types for Loader(). I don't quite understand the mechanism behind this. Of course, I see here that you have lists and instances of write and read, but I don't understand the loaders using the Loader() class.

Reply:
They are not duplicated for data modeling resources, and soon for the rest. This is the preparation I mentioned in the description. It is useful to distinguish between read and write classes when we want to do comparisons. In addition, we should always load from write classes and not read classes.

Reply:
Ok